src/performance_profiles.jl

```diff
@@ -129,3 +129,45 @@ function performance_profile(b::AbstractBackend,
   (x_plot, y_plot, max_ratio) = performance_profile_data(T, logscale=logscale, sampletol=sampletol, drawtol=drawtol)
   performance_profile_plot(b, x_plot, y_plot, max_ratio, xlabel, ylabel, labels, title, logscale; kwargs...)
 end
+"""
+    performance_profile(b,stats, cost)
+Produce a performance profile comparing solvers in `stats` using the `cost` function.
+Inputs:
+- `b :: AbstractBackend`: the backend used to produce the plot.
+- `stats::Dict{Symbol,DataFrame}`: pairs of `:solver => df`;
+- `cost::Function`: cost function applyed to each `df`. Should return a vector with the cost of solving the problem at each row;
+- if the cost is zero for at least one problem, all costs will be shifted by 1;
+- if the solver did not solve the problem, return Inf or a negative number.
+Examples of cost functions:
+- `cost(df) = df.elapsed_time`: Simple `elapsed_time` cost. Assumes the solver solved the problem.
+- `cost(df) = (df.status .!= :first_order) * Inf + df.elapsed_time`: Takes the status of the solver into consideration.
+"""
```

Comment:
Could we use the same docstring format as in the existing `performance_profile`?